OTVision/helpers/files.py

```diff
@@ -282,13 +285,50 @@ def denormalize_bbox(
         keys_width = ["x", "w"]
     if keys_height is None:
         keys_height = ["y", "h"]
-    if otdict["metadata"]["det"]["normalized"]:
-        direction = "denormalize"
-        otdict = _normal_transformation(otdict, direction, keys_width, keys_height)
-        otdict["metadata"]["det"]["normalized"] = False
-        log.debug("Dict denormalized")
-    else:
-        log.debug("Dict was already denormalized")
+    log.debug("Denormalize frame wise")
+    otdict = _denormalize_transformation(otdict, keys_width, keys_height, metadata)
+    return otdict
+
+
+# TODO: Type hint nested dict during refactoring
+def _denormalize_transformation(
```

Comment:
Denormalization and normalization have been separated again. The update of metadata is different in both methods. Therefore, the separation was necessary.

Metadata will now be read by frame, as frames belong to different video files.